src/visitors.rs

```diff
@@ -112,6 +121,33 @@ impl<'a, 'b: 'a, 'tcx: 'b> MirVisitor<'a, 'b, 'tcx> {
         refined_val
     }
 
+    /// For PathSelector::Deref, lookup the reference value using the qualifier, and then dereference that.
+    /// Otherwise return None.
+    fn try_to_deref(&mut self, path: &Path) -> Option<AbstractValue> {
+        match path {
```

Comment:
Not sure if this is actually an improvement or not, but at least consider:

```
if let Path::QualifiedPath { ref qualifier, ref selector } = **selector {
  if let PathSelector::Deref = selector {
    let ref_val = self.lookup_path_and_refine_result((**qualifier).clone());
     return Some(self.dereference(ref_val))
  }
}
None

Reply:
**selector should be path and selector should be **selector, but other than that, very nice! I am still learning Box magic. Thanks.